design/KEPS/multicluster/proposal.md

```diff
@@ -0,0 +1,160 @@
+# Multicluster support
+
+1. [Summary](#summary)
+2. [Motivation](#motivation)
+   1. [Goals](#goals)
+   2. [Non-Goals](#nongoals)
+3. [Solution](#solution)
+   1. [Other solutions](#othersolutions)
+4. [Roadmap](#roadmap)
+
+# Summary
+
+Multi-cluster support for Kiali. A “single pane of glass” for your istio service mesh.
+
+[github discussions](https://github.com/kiali/kiali/discussions/categories/multicluster)
+
+# Motivation
+
+It is now common for meshes to span across multiple clusters and yet Kiali only partially supports multi-cluster deployments today. Users should be able to visualize their workloads on the graph across clusters, dig into details via the workload/app/service pages, receive validations for their istio config etc. across their whole mesh from a single Kiali.
+
+## Goals
+
+- Multi-cluster mesh support - single pane of glass for your entire mesh.
+- Support multi-primary and primary-remote Istio deployment models.
+- Support RBAC authentication strategy for multi-cluster deployments.
+
+## Non-goals
+
+- Custom aggregation of all Kiali’s datasources across clusters. There are other tools and initiatives for aggregating metrics/logs/traces such as otel and this does not need to be handled by Kiali.
+- Ambient is not going to be touched in this effort.
+- Multi-mesh support - multi-cluster is focused on single mesh support only but shouldn’t prevent Kiali from supporting multi-mesh in the future.
+
+# Solution
+
+A single Kiali configured to access each Kubernetes cluster that is part of the mesh. Kiali will require Metrics/traces to be aggregated in a centralized datastore that Kiali can query.
+
+In a multicluster deployment, Kiali will only support the OIDC auth strategy for now.
+
+## Deployment models
+
+In each of the deployment models, these are the requirements for a centralized Kiali
```

Comment:
Establishing a more definitive name would be helpful. Istio has something similar with the "deployments models" e.g. "primary-remote". I went with "centralized" because I thought it somewhat described how Kiali would be deployed with one centralized Kiali. It is a vague term though. Maybe there's something less vague than "centralized" but more descriptive of the deployment model than "Multi-Cluster"?

Reply:
I don't have a better name.  I had thought maybe we would mimic Istio and could use Kiali Multi-Primary and Kiali Primary-Remote, but that may be confusing because we're not realy talking about the same thing, as we may apply this same deployment model in both of those istio models.  We really are talking about a single Kiali communicating with multiple Kube clusters, and so Kiali Multi-Cluster was the best thing I could think of.

I don't like "centralized" because it's vague and it possibly infers some sort of physical location, when that's not really the case.

Reply:
Someone could deploy multiple Kialis for greater redundancy across clusters but they would be independent of one another and would need to be configured separately. This is also not a requirement for Kiali MC. 

> I don't like "centralized" because it's vague and it possibly infers some sort of physical location, when that's not really the case.

"Single pane of glass" comes to mind but that's not very succinct and is probably more vague than "centralized".

Outside of this proposal, I'd prefer the more verbose Kiali Multi-Cluster over Kiali MC.